### PR TITLE
AET-203 Change stack definition to use the version of the agent corresponding to the current pipeline by default

### PR DESCRIPTION
**Pull request by KevinFairise2**
### What does this PR do?

By default the new e2e test will now use the version of the agent built in the pipeline

### Motivation

Running test toward the newly built agent in the pipeline

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: test/new-e2e/utils/e2e/stack_definition.go [hunk 1]
```diff
@@ -83,6 +85,10 @@ func AgentStackDef(vmParams []ec2params.Option, agentParameters ...agentparams.O
 				return nil, err
 			}
 
+			if pipelineId, ok := os.LookupEnv("CI_PIPELINE_ID"); ok {
```
Comment (ogaca-dd) [inline test/new-e2e/utils/e2e/stack_definition.go:88]: Pipeline ID is an internal data that should not be exposed to the user. It should be injected through the config map. See [this](https://github.com/DataDog/datadog-agent/blob/7.46.0/test/new-e2e/runner/configmap.go#L74-L106).

**Review by ogaca-dd — CHANGES_REQUESTED** [review]

**Review by ogaca-dd — APPROVED** [review]
LGTM

**Review by chouetz — APPROVED** [review]

**Comment by pducolin** [comment]
@L3n41c does it make sense to extend this to cluster and docker tests ? It would require setting `ddagent:clusterAgentFullImagePath` and `ddagent:fullImagePath` from `ddagent:pipeline_id`, plus would need adding a `ddagent:commit_sha`. Some dev on `test-infra` and on the load config part, on the pros side we would get tests running by default on a pipeline's build in any case

**Comment by L3n41c** [comment]
@pducolin For containers tests, `ddagent:clusterAgentFullImagePath` and `ddagent:fullImagePath` are already set from the pipeline ID here: https://github.com/DataDog/datadog-agent/blob/3ff5c025fb32e0d8c40be0f3a14e5706dfb6c676/.gitlab/e2e.yml#L179

**Pull request merged**